test/lld/em_asm_mainmodule_otherglobals.wat

```diff
@@ -0,0 +1,93 @@
+(module
```

Comment:
For these lld test we normally check in the C source code alongside the test and use `generate_lld_tests.py` to regenerate when we need to.

Reply:
Oh... there is no C source for this, I took pieces from other testcases and crafted it by hand...

Reply:
I don't think a C source would help in this case, but that's good to know.

Reply:
I guess what I mean is I'd like to avoid large chucks of test code like this checked without understanding where it comes from.

Also, I'm not sure how em_asm is relavant here.. wouldn't a simpler test just be to do something like:  `int foo = 0; int& bar = &foo`

Reply:
i.e. you don't em_asm to generate relocation against some global data.

Reply:
Yeah, the em_asm isn't necessary. It was just convenient to start from one of those testcases, remove the unnecessary bits, and add the new ones.

Makes sense to simplify testcases as much as possible. I reworked it now to avoid any irrelevant EM_ASM stuff, and it's much smaller. It's basically a minimal wat that has three `GOT` imports, and we want to see that only 2 of them are in the output.